cpp-rt.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef CPPRT_HPP
+#define CPPRT_HPP
+
+#include <stddef.h>
+#include <malloc.h>
+#include <cstdint>
+#include <cassert>
+
+#define FATAL(s) { fprintf(stderr, s); exit(1); }
```

Comment:
Might be nitpicking, but any chance we can make this not a macro?

Reply:
yep